Assets/LeapMotion/Scripts/DataStructures/RingBuffer.cs

```diff
@@ -1,68 +1,110 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
-using UnityEngine;
 
-public class RingBuffer<T> {
+namespace Leap.Unity {
 
-  private T[] arr;
-  private int firstIdx = 0;
-  private int lastIdx = 0;
+  public class RingBuffer<T> {
 
-  public RingBuffer(int bufferSize) {
-    arr = new T[bufferSize];
-  }
+    private T[] arr;
+    private int firstIdx = 0;
+    private int lastIdx = 0;
 
-  public int Length {
-    get {
-      if (lastIdx == -1) return 0;
-      int diff = (lastIdx + 1) - firstIdx;
-      if (diff <= 0) return diff + arr.Length;
-      return diff;
+    public RingBuffer(int bufferSize) {
+      arr = new T[bufferSize];
     }
-  }
 
-  public bool IsFull {
-    get { return Length == arr.Length; }
-  }
+    public int Length {
+      get {
+        if (lastIdx == -1) return 0;
+        int diff = (lastIdx + 1) - firstIdx;
+        if (diff <= 0) return diff + arr.Length;
+        return diff;
+      }
+    }
 
-  public void Clear() {
-    lastIdx = -1;
-    firstIdx = 0;
-  }
+    public bool IsFull {
+      get { return Length == arr.Length; }
+    }
 
-  public void Add(T t) {
-    if (IsFull) {
-      firstIdx += 1;
-      firstIdx %= arr.Length;
+    public void Clear() {
+      lastIdx = -1;
+      firstIdx = 0;
     }
-    lastIdx += 1;
-    lastIdx %= arr.Length;
 
-    arr[lastIdx] = t;
-  }
+    public void Add(T t) {
+      if (IsFull) {
+        firstIdx += 1;
+        firstIdx %= arr.Length;
+      }
+      lastIdx += 1;
+      lastIdx %= arr.Length;
 
-  /// <summary>
-  /// Oldest element is at index 0, youngest is at Length - 1.
-  /// </summary>
-  public T Get(int idx) {
-    return arr[(firstIdx + idx) % arr.Length];
-  }
+      arr[lastIdx] = t;
+    }
 
-  public T GetLatest() {
-    return Get(Length - 1);
-  }
+    /// <summary>
+    /// Oldest element is at index 0, youngest is at Length - 1.
+    /// </summary>
+    public T Get(int idx) {
+      return arr[(firstIdx + idx) % arr.Length];
+    }
 
-  public T GetOldest() {
-    return Get(0);
-  }
+    public T GetLatest() {
+      return Get(Length - 1);
+    }
 
-  public void Set(int idx, T t) {
-    int actualIdx = (firstIdx + idx) % arr.Length;
-    arr[actualIdx] = t;
-  }
+    public T GetOldest() {
+      return Get(0);
+    }
+
+    public void Set(int idx, T t) {
+      int actualIdx = (firstIdx + idx) % arr.Length;
+      arr[actualIdx] = t;
+    }
+
+    public void SetLatest(T t) {
+      Set(Length - 1, t);
+    }
+
+    public RingBufferEnumerator GetEnumerator() {
+      return new RingBufferEnumerator(this);
+    }
+
+    public class RingBufferEnumerator : IEnumerator<T>, System.Collections.IEnumerator {
```

Comment:
If you make RingBufferEnumerator a struct you can use foreach without creating garbage